docs/src/main/asciidoc/_developer-guide/JTA_Reference.adoc

```diff
@@ -0,0 +1,394 @@
+or you can use annotations.
+The https://javaee.github.io/javaee-spec/javadocs/javax/transaction/Transactional.html[`@Transactional`]
+defines transaction boundaries
+(start of the method starts the transaction, while the transaction is finished at its end).
+If the method finishes sucessfully the transaction is committed.
+If the transaction ends with `RuntimeException` then it's rolled-back.
+
+When you define the method to be `@Transactional` you defines behaviour
+of incoming (or non-existent) transactional context.
+Please refer to documentation for
+https://javaee.github.io/javaee-spec/javadocs/javax/transaction/Transactional.TxType.html[`Transactional.TxType`].
+
+It's important to note that the `TxType` influences the transaction management
+*only* when called with the managed CDI bean. If you call the method directly -
+without the CDI container wraps the invocation then the `TxType` has no effect.
+
+[source,java,options="nowrap"]
+----
+@RequestScope
+public class MyCDIBean {
+  @Inject
+  MyCDIBean myBean;
+
+  @Transactional(TxType.REQUIRED)
+  public void mainMethod() {
+    // CDI container does not wrap the invocation
+    // no new transaction is started
+    innerFunctionality();
+
+    // CDI container starts a new transaction
+    // the method uses TxType.REQUIRES_NEW and is called from the CDI bean
+    myBean.innerFunctionality();
+  }
+
+  @Transactional(TxType.REQUIRES_NEW)
+  private void innerFunctionality() {
+    // some business logic
+  }
+}
+----
+
+NOTE: the exception handling could be influenced by attributes
+      `rollbackOn` and `dontRollbackOn` (of the `@Transactional` annotation)
+
+WARNING: if you use the `@Transactional` for managing transactions boundaries
+         you won't be permitted to use the `UserTransaction` methods.
+         If you do so you can expect an runtime exception being thrown.
+
+The CDI introduces as well a new scope for use with transactions -
+https://javaee.github.io/javaee-spec/javadocs/javax/ejb/SessionContext.html[`@TransactionScoped`].
+
+[[in-ejbs]]
+=== Transactions in EJBs
+
+Transaction management in EJB is administered with two annotations:
+https://javaee.github.io/javaee-spec/javadocs/javax/ejb/TransactionManagement.html[`@TransactionManagement`]
+ https://javaee.github.io/javaee-spec/javadocs/javax/ejb/TransactionAttribute.html[`@TransactionAttribute`].
+
+NOTE: you can define the same behaviour when you use the `ejb-jar.xml`
+      descriptor instead of the annotations
+
+The WildFly provides specific annotation `org.jboss.ejb3.annotation.TransactionTimeout`
+which gives you chance to change transaction timeout for particular bean/method.
+
+NOTE: when you use message driven bean then the `@TransactionTimeout` does not work
+      and you need to define the timeout through the `@ActivationConfigProperty`:
+      `@ActivationConfigProperty(propertyName="transactionTimeout", propertyValue="1")`
+
+By default EJB method is annotated like
```

Comment:
Is the meaning here...

"By default the behavior of an EJB method is as if it were annotated like"

?

I read the sentence at L178 as meaning we somehow annotate the method, e.g. bytecode manipulation or something.

Reply:
agree, it could be misguiding and I failed to express the point here. I will try better :-)

Reply:
Nice!